### DISCUSSION TITLE
type anotation error after upgrade of fastapi to latests,

### DISCUSSION THREAD

**Discussion by michelkluger** [laurentS/slowapi#203] [Q&A]
apollo/api.py:75: error: Argument 2 to "add_exception_handler" of "Starlette" has incompatible type "Callable[[Request, RateLimitExceeded], Response]"; expected "Callable[[Request, Exception], Response | Awaitable[Response]] | Callable[[WebSocket, Exception], Awaitable[None]]" 

comes from line 
 app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)
 
 how can I fix this?
 
  starlette-0.37.2
  fastapi-0.111.0

**Comment by ecly** [comment]
@michelkluger most likely you're annotating the the function `_rate_limit_exceeded_handler` to only handle `RateLimitExceeded` exception type, which will make it incompatible with expected Callable signature.

So the "fix" in this case would be annotate your function as `def _rate_limit_exceeded_handler(request: Request, exception: Exception)` instead (or just ignore the error, as you know it will only get called with `RateLimitExceeded` exceptions).
